### PR TITLE
Add migrations argument to load_text and dump_text

### LINKED ISSUE #19: Text format for migrations
There is currently no support for reading/writing migrations as text. 

Should be a straightforward update.

### PR DESCRIPTION
**Pull request by szhan**
Fixes #19

### CODE REVIEW

**Comment by szhan** [comment]
Yikes, I just updated `dump_text`. It seems that we may want to add a new `tskit` CLI subcommand to print out `migrations`.

**Comment by szhan** [comment]
I just consulted @benjeffery about this. It is a bit late for me to separate `load_text()` and `dump_text()` into two PRs. But I will create a separate PR to implement a subcommand to print out `migrations`.

**Comment by benjeffery** [comment]
Is this all ready for review?

File: python/tskit/trees.py [hunk 1]
```diff
@@ -3525,13 +3525,83 @@ def parse_populations(
...
+    """
+    Parse the specified file-like object containing a whitespace delimited
+    description of a migration table and returns the corresponding
+    :class:`MigrationTable` instance.
+
+    See the :ref:`migration text format <sec_migration_text_format>` section
+    for the details of the required format and the
+    :ref:`migration table definition <sec_migration_table_definition>` section
+    for the required properties of the contents. Note that if the ``time`` column
+    is missing its entries are filled with ``UNKNOWN_TIME``.
```
Comment (benjeffery) [inline python/tskit/trees.py:3540]:
```suggestion
    is missing its entries are filled with :data:`UNKNOWN_TIME`.
```

**Review by benjeffery — APPROVED** [review]
Nice!
One doc nit.

**Pull request merged**